### PR TITLE
Added font style support

### PR DESCRIPTION
**Pull request by Pryadka**
All changes were done according to discussion https://github.com/rism-ch/verovio/issues/1460

- read character size information for bold and italic fonts;
- apply corresponding sizes for characters;

### CODE REVIEW

File: src/verse.cpp [hunk 1]
```diff
@@ -175,7 +176,11 @@ int Verse::AdjustSylSpacing(FunctorParams *functorParams)
 
     // Use the syl because the content bounding box of the verse might be invalid at this stage
     int overlap = params->m_lastSyl->GetContentRight() - (firstSyl->GetContentLeft() + xShift);
-    overlap += lastSyl->CalcConnectorSpacing(params->m_doc, params->m_staffSize);
+    overlap += params->m_lastSyl->CalcConnectorSpacing(params->m_doc, params->m_staffSize);
+
+    if (params->m_lastSyl->GetUuid() == "syl-0000001599481773") {
+        int d = params->m_doc->GetDrawingUnit(params->m_staffSize);
+    }
```
Comment (lpugin) [inline src/verse.cpp:183]: ??

File: src/verse.cpp [hunk 2]
```diff
@@ -138,6 +138,7 @@ int Verse::AdjustSylSpacing(FunctorParams *functorParams)
             Syl *syl = dynamic_cast<Syl *>(*iter);
             assert(syl);
             syl->SetDrawingXRel(previousSylShift);
+            auto t = syl->GetUuid();
```
Comment (lpugin) [inline src/verse.cpp:141]: Don't use `auto` in such cases

File: include/vrv/devicecontextbase.h [hunk 3]
```diff
@@ -102,8 +102,8 @@ class FontInfo {
     {
         m_pointSize = 0;
         m_family = 0; // was wxFONTFAMILY_DEFAULT;
-        m_style = FONTSTYLE_NONE;
-        m_weight = FONTWEIGHT_NONE;
+        m_style = FONTSTYLE_normal;
+        m_weight = FONTWEIGHT_normal;
```
Comment (lpugin) [inline include/vrv/devicecontextbase.h:106]: Doesn't this mean that `normal` font style / weight will be added to every single text element in the SVG? I think we want to avoid this. @rettinghaus ?

File: src/view_element.cpp [hunk 4]
```diff
@@ -1399,6 +1399,9 @@ void View::DrawSyl(DeviceContext *dc, LayerElement *element, Layer *layer, Staff
         dc->SetFont(m_doc->GetDrawingLyricFont(staff->m_drawingStaffSize));
     }
 
+    dc->GetFont()->SetStyle(syl->HasFontstyle() ? syl->GetFontstyle() : data_FONTSTYLE::FONTSTYLE_normal);
+    dc->GetFont()->SetWeight(syl->HasFontweight() ? syl->GetFontweight() : data_FONTWEIGHT::FONTWEIGHT_normal);
```
Comment (lpugin) [inline src/view_element.cpp:1403]: No need to prefix `FONTSTYLE_normal` and `FONTWEIGHT_normal`
Again, do we always want / need the font style / weigh to be set to normal?

File: src/view_text.cpp [hunk 5]
```diff
@@ -356,6 +356,10 @@ void View::DrawText(DeviceContext *dc, Text *text, TextDrawingParams &params)
 
     dc->StartTextGraphic(text, "", text->GetUuid());
 
+    if(dc->GetFont()->GetWeight() != FONTWEIGHT_NONE && dc->GetFont()->GetStyle() != FONTSTYLE_NONE) {
```
Comment (lpugin) [inline src/view_text.cpp:359]: Please use `HasWeight()` and `HasStyle()` instead

File: src/verse.cpp (hunk 1 above)
Comment (Pryadka) [reply src/verse.cpp:183]: Sorry, this is about my further work. I'll remove it.

File: src/view_text.cpp (hunk 5 above)
Comment (Pryadka) [reply src/view_text.cpp:359]: `FontInfo `doesn't have such methods. Should I implement them?

File: include/vrv/devicecontextbase.h (hunk 3 above)
Comment (rettinghaus) [reply include/vrv/devicecontextbase.h:106]: Definitely! We have Verovio's "house style" set in the svg style element. So `@fontstyle` should only be added, if it is used in the input file.

Comment (Pryadka) [reply include/vrv/devicecontextbase.h:106]: Ok. I will care about it.

File: src/view_element.cpp (hunk 4 above)
Comment (Pryadka) [reply src/view_element.cpp:1403]: Currently if style/weight is not specified in the input, we assume that it is normal. Normal means default. From rendering point of view a font type cant be undefined. And for sure we can omit these attributes in the output if they are _normal.

File: src/verse.cpp (hunk 2 above)
Comment (Pryadka) [reply src/verse.cpp:141]: The same. Will be removed.

File: src/view_text.cpp (hunk 5 above)
Comment (lpugin) [reply src/view_text.cpp:359]: Ah, sorry, I thought it was checking against an attribute class. You can leave it like this

**Comment by Pryadka** [comment]
@lpugin  Please review it again.

File: include/vrv/vrv.h [hunk 6]
```diff
@@ -128,35 +130,42 @@ bool Check(Object *object);
...
     /** The path to the resources directory (e.g., for the svg/ subdirectory with fonts as XML */
     static std::string m_path;
     /** The loaded SMuFL font */
-    static std::map<wchar_t, Glyph> m_font;
+    static GlyphMap m_font;
     /** A text font used for bounding box calculations */
-    static std::map<wchar_t, Glyph> m_textFont;
+    static GlyphTextMap m_textFont;
+    static StyleAttributes m_currentStyle;
+    static const StyleAttributes k_defaultStyle;
```
Comment (lpugin) [inline include/vrv/vrv.h:168]: Sorry, this was my mistake, but static member as normally prefixed with `s_` and not `m_`. Could you change them? Thanks!

File: src/verse.cpp [hunk 7]
```diff
@@ -175,7 +175,7 @@ int Verse::AdjustSylSpacing(FunctorParams *functorParams)
 
     // Use the syl because the content bounding box of the verse might be invalid at this stage
     int overlap = params->m_lastSyl->GetContentRight() - (firstSyl->GetContentLeft() + xShift);
-    overlap += lastSyl->CalcConnectorSpacing(params->m_doc, params->m_staffSize);
+    overlap += params->m_lastSyl->CalcConnectorSpacing(params->m_doc, params->m_staffSize);
```
Comment (lpugin) [inline src/verse.cpp:178]: What is the reason for this change? Was it wrong?

File: src/vrv.cpp [hunk 8]
```diff
@@ -100,13 +120,34 @@ Glyph *Resources::GetGlyph(wchar_t smuflCode)
...
+}
+
 Glyph *Resources::GetTextGlyph(wchar_t code)
 {
-    if (!m_textFont.count(code)) return NULL;
-    return &m_textFont[code];
+    GlyphMap *currentMap = &m_textFont[m_textFont.count(m_currentStyle) != 0 ? m_currentStyle : k_defaultStyle];
+
+    if (currentMap->count(code) == 0)
+            return NULL;
```
Comment (lpugin) [inline src/vrv.cpp:145]: On one line or with {}

File: src/verse.cpp (hunk 7 above)
Comment (Pryadka) [reply src/verse.cpp:178]: Logical. lastSyl is current last syllable. But in this line we calculate space between current syllable and previous syllable. This fix is not related to the current task.

File: src/vrv.cpp (hunk 8 above)
Comment (Pryadka) [reply src/vrv.cpp:145]: Done.

File: src/verse.cpp (hunk 7 above)
Comment (lpugin) [reply src/verse.cpp:178]: Correct. Leave it

File: include/vrv/vrv.h (hunk 6 above)
Comment (Pryadka) [reply include/vrv/vrv.h]: Done
